Tests/js/views/services/assets/ez-trashviewservice-tests.js

```diff
@@ -942,9 +957,141 @@ YUI.add('ez-trashviewservice-tests', function (Y) {
         },
     });
 
+    restoreItemsTest = new Y.Test.Case({
+        name: "eZ Trash View Service restore trash items tests",
+
+        setUp: function () {
+            this.app = new Mock();
+            this.capi = new Mock();
+
+            this.contentServiceMock = new Mock();
+            Mock.expect(this.contentServiceMock, {
+                method: "emptyTrash",
+                args: [Mock.Value.Function],
+                run: function (callback) {
+                    callback();
+                },
+            });
+
+            this.capiMock = new Mock();
+            Mock.expect(this.capiMock, {
+                method: "getContentService",
+                args: [],
+                returns: this.contentServiceMock,
+            });
+
+            this.service = new Y.eZ.TrashViewService({
+                app: this.app,
+                capi: this.capiMock,
+                trashItems: [1, 2, 3],
+            });
+        },
+
+        tearDown: function () {
+            this.service.destroy();
+            delete this.service;
+        },
+
+        _createTrashItem: function (id, error) {
+            return {
+                restore: function (loadOptions, callback) {
```

Comment:
a Mock would be more suited here.
Also a test where you assert that `location#restore` is called is missing.